modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v3_20_0/ObjectFieldUpgradeProcess.java

```diff
@@ -0,0 +1,141 @@
+			StringBundler.concat(
+			"select ObjectDefinition.objectDefinitionId, ",
+				"ObjectDefinition.companyId, ObjectDefinition.userName, ",
+			"ObjectDefinition.userId from ObjectDefinition where ",
+			"ObjectDefinition.system_ = 0"));
+
+			ResultSet resultSet = preparedStatement1.executeQuery()) {
+
+			while (resultSet.next()) {
+				PreparedStatement preparedStatement2 =
+					AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+						connection,
+						StringBundler.concat(
+							"insert into ObjectField (mvccVersion, uuid_, ",
+							"objectFieldId, companyId, userId, userName, ",
+							"createDate, modifiedDate, externalReferenceCode, ",
+							"listTypeDefinitionId, objectDefinitionId, ",
+							"businessType, dbColumnName, dbTableName, dbType, ",
+							"defaultValue, indexed, indexedAsKeyWord, ",
+							"indexedLanguageId, label, name, ",
+							"relationshipType, required, state_, system_) ",
+							"values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ",
+							"?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)"));
+
+				preparedStatement2.setLong(1, 0);
+
+				String uuid = _portalUUID.generate();
+
+				preparedStatement2.setString(2, uuid);
+
+				preparedStatement2.setLong(3, increment());
+
+				long companyId = resultSet.getLong("companyId");
+
+				preparedStatement2.setLong(4, companyId);
+
+				preparedStatement2.setLong(5, resultSet.getLong("userId"));
+				preparedStatement2.setString(
+					6, resultSet.getString("userName"));
+
+				Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+
+				preparedStatement2.setTimestamp(7, timestamp);
+				preparedStatement2.setTimestamp(8, timestamp);
+
+				preparedStatement2.setString(9, uuid);
+				preparedStatement2.setLong(10, 0);
+				preparedStatement2.setLong(
+					11, resultSet.getLong("objectDefinitionId"));
+				preparedStatement2.setString(
+					12, ObjectFieldConstants.BUSINESS_TYPE_TEXT);
+				preparedStatement2.setString(
+					13,
+					ObjectEntryTable.INSTANCE.externalReferenceCode.getName());
+				preparedStatement2.setString(14, "ObjectEntry");
+				preparedStatement2.setString(
+					15, ObjectFieldConstants.DB_TYPE_STRING);
+				preparedStatement2.setNull(16, 0);
+				preparedStatement2.setInt(17, 0);
+				preparedStatement2.setInt(18, 0);
+				preparedStatement2.setNull(19, 0);
+
+				Locale defaultLocale = LocaleUtil.fromLanguageId(
+					UpgradeProcessUtil.getDefaultLanguageId(companyId));
+
+				preparedStatement2.setString(
+					20,
+					LocalizationUtil.getXml(
+						new LocalizedValuesMap() {
+							{
+								put(
+									defaultLocale,
+									LanguageUtil.get(
+										defaultLocale,
+										"external-reference-code"));
+							}
+						},
+						"Label"));
+
+				preparedStatement2.setString(21, "externalReferenceCode");
+				preparedStatement2.setNull(22, 0);
+				preparedStatement2.setInt(23, 0);
+				preparedStatement2.setInt(24, 0);
+				preparedStatement2.setInt(25, 1);
```

Comment:
Hi @p-albuquerque,

system_, state_, required and other boolean fields should be defined as boolean, you must use:

`preparedStatement2.setBoolean`